pkg/quota/v1/evaluator/core/resource_claims.go

```diff
@@ -99,10 +117,102 @@ func (p *claimEvaluator) MatchingResources(items []corev1.ResourceName) []corev1
 			strings.HasSuffix(string(item), corev1.ResourceClaimsPerClass /* by device class */) {
 			result = append(result, item)
```

Comment:
can `continue` if we already appended the item here and not do more checks below